### DISCUSSION TITLE
Slow when developing locally?

### DISCUSSION THREAD

**Discussion by RomkaLTU** [planetscale/discussion#461] [General]
Hello, maybe I'm doing something wrong but when developing locally (Laravel Valet, Laravel) it feels unresponsive/slow it's always taking about 2 seconds till the first bite transfer:

![CleanShot 2023-05-08 at 21 12 27](https://user-images.githubusercontent.com/10167743/236899306-fd18b7ca-056d-4baf-b452-244e7706dbc1.png)

Maybe I'm doing something wrong? My internet connection is not an issue here 100%.

**Comment by orware** [comment]
Hello @RomkaLTU,

This could be due to a variety of factors.

One of the first questions we'd ask about here would be what the overall distance between your machine is and the PlanetScale region your branch is located in, as the physical distance could be quite impactful there.

DNS could be another factor, depending on the answer to the above question since sometimes depending on the ISP the geolocation routing may not go through the expected most efficient path for our generic `aws.connect.psdb.cloud` hostname so in some of those cases you may have to utilize a region specific hostname to workaround that issue. A recent fly.io thread that discusses that very situation can be found here:
https://community.fly.io/t/planetscale-large-latency/12576/1 

If it isn't network latency due to physical distance or DNS issues, and you can see the queries themselves that might be running for the above page example you shared are all fast from within your Insights area then there might be something else we can troubleshoot together.

If you continue to experience slowness, and the thoughts above don't seem to be the issue, please reach out via https://support.planetscale.com/ and we can work together more closely on it!
